accurest-core/src/main/groovy/io/coderate/accurest/dsl/Request.groovy

```diff
@@ -0,0 +1,172 @@
+    }
+
+    class Headers {
+        private Map<String, WithValuePattern> headers = [:]
+
+        WithValuePattern header(String headerName) {
+            WithValuePattern withValuePattern = new WithValuePattern()
+            headers[headerName] = withValuePattern
+            return withValuePattern
+        }
+
+        Set<Map.Entry<String, WithValuePattern>> entries() {
+            return headers.entrySet()
+        }
+    }
+
+/**
+ *
+ * From Wiremock
+ private Map<String, ValuePattern> queryParamPatterns;
+ private List<ValuePattern> bodyPatterns;
+
+ private String equalToJson;
+ private String equalToXml;
+ private String matchesXPath;
+ private JSONCompareMode jsonCompareMode;
+ private String equalTo;
+ private String contains;
+ private String matches;
+ private String doesNotMatch;
+ private Boolean absent;
+ private String matchesJsonPath;
+
+ */
+    class WithValuePattern {
+        NoOpCustomizableProperty<String> equalToJson
+        NoOpCustomizableProperty<String> equalToXml
+        StringCustomizableProperty matchesXPath
+        NoOpCustomizableProperty<JSONCompareMode> jsonCompareMode
+        NoOpCustomizableProperty<String> equalTo
+        StringCustomizableProperty contains
+        StringCustomizableProperty matches
+        StringCustomizableProperty doesNotMatch
+        CustomizableProperty<Boolean, Boolean> absent
+        StringCustomizableProperty matchesJsonPath
+
+//        void equalToJson(String equalToJson) {
+//            this.equalToJson = equalToJson
+//        }
+//
+        void equalTo(String equalTo) {
+            this.equalTo = new NoOpCustomizableProperty(equalTo)
+        }
+
+//        void equalToXml(String equalToXml) {
+//            this.equalToXml = equalToXml
+//        }
+
+        void matches(@DelegatesTo(StringCustomizableProperty) Closure closure) {
+            StringCustomizableProperty placeholderHaving = new StringCustomizableProperty()
+            this.matches = placeholderHaving
+            delegateToClosure(closure, placeholderHaving)
+        }
+
+        void doesNotMatch(@DelegatesTo(StringCustomizableProperty) Closure closure) {
+            StringCustomizableProperty placeholderHaving = new StringCustomizableProperty()
+            this.doesNotMatch = placeholderHaving
+            delegateToClosure(closure, placeholderHaving)
+        }
+
+        void contains(@DelegatesTo(StringCustomizableProperty) Closure closure) {
+            StringCustomizableProperty placeholderHaving = new StringCustomizableProperty()
+            this.contains = placeholderHaving
+            delegateToClosure(closure, placeholderHaving)
+        }
+
+//        void jsonCompareMode(JSONCompareMode jsonCompareMode) {
+//            this.jsonCompareMode = jsonCompareMode
+//        }
+
+    }
+
+    enum JSONCompareMode {
+        STRICT, LENIENT, NON_EXTENSIBLE, STRICT_ORDER;
+    }
+
+    private static <T> void delegateToClosure(@DelegatesTo(T) Closure closure, T delegate) {
```

Comment:
True! :) But from what I remember we're using local versoins of groovy and gradle in the plugin? :)